File: ecos-records/src/main/java/ru/citeck/ecos/records2/exception/RecordsSourceNotFoundException.java
package ru.citeck.ecos.records2.exception;

import ru.citeck.ecos.records2.source.dao.RecordsDao;

public class RecordsSourceNotFoundException extends RecordsException {

    private final Class<? extends RecordsDao> type;
    private final String sourceId;

    public RecordsSourceNotFoundException(String sourceId, Class<? extends RecordsDao> type) {
        super("Source is not found with id '" + sourceId + "' and "
                + "type '" + (type != null ? type.getName() : "null") + "'");

        this.sourceId = sourceId;
        this.type = type;
    }

    public String getSourceId() {
        return sourceId;
    }

    public Class<? extends RecordsDao> getType() {
        return type;
    }
}

File: ecos-records/src/main/java/ru/citeck/ecos/records3/exception/RecordsSourceNotFoundException.kt
package ru.citeck.ecos.records3.exception

class RecordsSourceNotFoundException(val sourceId: String, val type: Class<*>?) : RecordsException(
    "Source is not found with id '" + sourceId + "' and " +
        "type '" + (if (type != null) type.name else "null") + "'"
)
